src/java/fr/paris/lutece/util/signrequest/IPAuthentificator.java

```diff
@@ -0,0 +1,72 @@
+package fr.paris.lutece.util.signrequest;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class IPAuthentificator implements RequestAuthenticator {
+	
+	private List<String> _listIPs;
+	private boolean isWhiteList;
+	
+	/**
+	 * Gets the list of IP adresses
+	 * 
+	 * @return the list of IP adresses
+	 */
+	public List<String> getIPs( ) {
+		return _listIPs;
+	}
+	
+	/**
+	 * Sets the list of IP adresses
+	 * 
+	 * @param list
+	 * 		The list of IP adresses
+	 */
+	public void setIPs( List<String> list ) {
+		this._listIPs = list;
+	}
+	
+	/**
+	 * Gets the boolean that defines if it is a white or black list
+	 * 
+	 * @return the boolean
+	 */
+	public boolean isWhiteList( ) {
+		return isWhiteList;
+	}
+
+	/**
+	 * Sets the boolean that defines if it is a white or black list
+	 * 
+	 * @param isWhiteList
+	 * 		The boolean
+	 */
+	public void setWhiteList( boolean isWhiteList ) {
```

Comment:
Method with boolean attribute might be confusing at call sites. Maybe declare an enum with ALLOW and BLOCK cases, and rename this method setMode for instance.

Reply:
I integrated an enum to allow & block instead of the boolean.
Thanks for the feedback